net/proxy_resolution/proxy_config_service_tor.cc

```diff
@@ -123,33 +124,12 @@ void ProxyConfigServiceTor::UpdateProxyURI(const std::string& uri) {
                                   config_valid);
 }
 
-// static
-std::string ProxyConfigServiceTor::CircuitIsolationKey(const GURL& url) {
-  // https://2019.www.torproject.org/projects/torbrowser/design/#privacy
-  //
-  //    For the purposes of the unlinkability requirements of this
-  //    section as well as the descriptions in the implementation
-  //    section, a URL bar origin means at least the second-level DNS
-  //    name.  For example, for mail.google.com, the origin would be
-  //    google.com.  Implementations MAY, at their option, restrict
-  //    the URL bar origin to be the entire fully qualified domain
-  //    name.
-  //
-  // In particular, we need not isolate by the scheme,
-  // username/password, port, path, or query part of the URL.
-  url::Origin origin = url::Origin::Create(url);
-  std::string domain = net::registry_controlled_domains::GetDomainAndRegistry(
-      origin.host(),
-      net::registry_controlled_domains::INCLUDE_PRIVATE_REGISTRIES);
-  if (domain.size() == 0)
-    domain = origin.host();
-  return domain;
-}
-
 void ProxyConfigServiceTor::SetNewTorCircuit(const GURL& url) {
   const HostPortPair old_host_port = proxy_server_.host_port_pair();
+
+  url::Origin url_origin = url::Origin::Create(url);
   const HostPortPair new_host_port(
-      CircuitIsolationKey(url),
+      net::NetworkIsolationKey(url_origin, url_origin).ToString(),
```

Comment:
ToString will give us "[origin] [origin]" , this is not what we what for username of socks5 config

Reply:
Yes, this was completely wrong and obviously the tests failed as well. Sorry for the sloppy first upload.

Just force-pushed the branch now and it's now passing all tests and working, PTAL again thanks!